lib/charms/livy.py

```diff
@@ -0,0 +1,90 @@
+from glob import glob
+from path import Path
+from subprocess import CalledProcessError, call
+import os
+import jujuresources
+from charmhelpers.core import hookenv
+from charmhelpers.core import host
+from charmhelpers.core import unitdata
+from charmhelpers.fetch.archiveurl import ArchiveUrlFetchHandler
+from jujubigdata import utils
+
+
+# Main Livy class for callbacks
+class Livy(object):
+    def __init__(self, dist_config):
+        self.dist_config = dist_config
+        self.resources = {
+            'livy': 'livy-%s' % utils.cpu_arch(),
+        }
+        self.verify_resources = utils.verify_resources(*self.resources.values())
+
+    def is_installed(self):
+          return unitdata.kv().get('livy.installed')
+
+    def install(self):
+        if self.is_installed():
+            return
+        jujuresources.install(self.resources['livy'],
+                              destination=self.dist_config.path('livy'),
+                              skip_top_level=False)
+
+        livy_bin = self.dist_config.path('livy') / 'bin'
+        with utils.environment_edit_in_place('/etc/environment') as env:
+            if livy_bin not in env['PATH']:
+                env['PATH'] = ':'.join([env['PATH'], livy_bin])
+            # Following classpath comes from `hadoop classpath` and should be fixed
+            hadoop_cp = '/etc/hadoop/conf:/usr/lib/hadoop/share/hadoop/common/lib/*:/usr/lib/hadoop/share/hadoop/common/*\
+:/usr/lib/hadoop/share/hadoop/hdfs:/usr/lib/hadoop/share/hadoop/hdfs/lib/*\
+:/usr/lib/hadoop/share/hadoop/hdfs/*:/usr/lib/hadoop/share/hadoop/yarn/lib/*\
+:/usr/lib/hadoop/share/hadoop/yarn/*:/usr/lib/hadoop/share/hadoop/mapreduce/lib/*\
+:/usr/lib/hadoop/share/hadoop/mapreduce/*:/usr/lib/hadoop/contrib/capacity-scheduler/*.jar'
```

Comment:
Minor nit: this could be formatted nicer using `':'.join([...])`.  Also, are directories in this list not naturally recursive?  That is, do we really need both `/blah/hdfs` and `/blah/hdfs/*`?